brendaleearmstrong/S4-Midterm-Server
# Refactor: standardize backend and improve models

Backend Standardization:
- Converted snake_case to camelCase in Java code
- Maintained snake_case for database columns
- Added proper response entities with HTTP status codes
- Improved error handling across controllers
- Added CORS support
- Standardized API endpoints

Model Updates:
- Mine:
  * Standardized field names
  * Added proper JPA annotations
  * Updated relationships
  
- Mineral:
  * Updated field names to camelCase
  * Added proper column mappings
  * Fixed relationship with MineMineral
  
- MineMineral:
  * Improved relationship mappings
  * Added proper validation
  * Fixed toString() implementation

Controller Improvements:
- Added proper @RequestMapping paths
- Standardized endpoint responses
- Improved error handling
- Added CORS configurations
- Added proper response status codes

Service Layer:
- Standardized method names
- Added proper validation
- Improved error handling
- Added transaction support

Repository Updates:
- Added proper annotations
- Fixed query method names
- Added necessary custom queries

API Changes:
- All endpoints now follow REST conventions
- Standardized response formats
- Improved error messages
- Added proper status codes

Breaking Changes:
- API request/response fields now use camelCase
- Updated endpoint paths
- Changed response formats

Testing Required:
- Verify all CRUD operations
- Test error handling
- Validate relationships
- Check API responses

Migration Notes:
- Update frontend to use new field names
- Update API calls to use new endpoints
- Review error handling implementation
